### PR TITLE
build: enable building for Fedora40 and Ubuntu 24.04

### PR DESCRIPTION
**Pull request by pstorz**
#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

### CODE REVIEW

**Review by arogge — CHANGES_REQUESTED** [review]
I have a few remarks concerning the CPM integration. But overall this looks good!

**Review by arogge — CHANGES_REQUESTED** [review]
move CPMAddPackage() calls into its own file and include in top-level CMakeLists.txt

File: core/cmake/BareosFindPrograms.cmake [hunk 1]
```diff
@@ -24,12 +24,10 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
...
 endif()
 
+# ps is required for systemtests
 find_program(PIDOF pidof)
-if(NOT PIDOF)
-  set(PIDOF "")
-endif()
 
-find_program(PS ps)
+find_program(PS ps REQUIRED)
```
[the remark this answers is not in the document]
Comment (pstorz) [reply core/cmake/BareosFindPrograms.cmake:30]: Done

**Comment by arogge** [comment]
Relates to #1850

**Review by arogge — APPROVED** [review]
This got a lot bigger than anticipated. But I think it is finally ready to be merged.

**Pull request merged**